torch/functional.py

```diff
@@ -55,6 +56,29 @@ def broadcast_tensors(*tensors):
     return torch._C._VariableFunctions.broadcast_tensors(tensors)
 
 
+def broadcast_to(tensor, shape):
+    r""" broadcast_to(tensor, shape) -> Tensor
+
+    Broadcast the given tensor to the given shape.
+
+    Args:
+        tensor (Tensor): tensor to broadcast.
+        shape (list(int)): broadcasted shape
+
+    Example::
+        >>> broadcast_to(torch.ones(3), (4, 3)).shape
+        torch.Size([4, 3])
+        >>> broadcast_to(torch.ones(1, 3), (4, 3)).shape
+        torch.Size([4, 3])
+    """
+    delta_len = tensor.dim() - len(shape)
+    if delta_len > 0:
+        # add singleton dimensions
+        tensor = tensor[[slice(None)] * delta_len]
+
+    return tensor.expand(shape)
```

Comment:
This function would never error out. For example:

```python
torch.broadcast_to(torch.randn(3), (3, 4))
```
would not be a valid broadcast.

I think using `torch._C._infer_size` here would be a good idea, as it errors out if the sizes are not broadcastable.